velox/functions/lib/Re2Functions.cpp

```diff
@@ -412,6 +425,47 @@ class OptimizedLikeWithMemcmp final : public VectorFunction {
       vector_size_t reducedPatternLength)
       : pattern_{pattern}, reducedPatternLength_{reducedPatternLength} {}
 
+  OptimizedLikeWithMemcmp(
+      StringView pattern,
+      vector_size_t reducedPatternLength,
+      std::vector<std::string> fixedPatterns)
+      : pattern_{pattern},
+        reducedPatternLength_{reducedPatternLength},
+        fixedPatterns_{fixedPatterns} {}
+
+  // Match all fixed patterns in an input string having fixed patterns
+  // interspersed with one or more '%' wildcard character streams.
+  bool matchFixedWithWildcard(const StringView input) const {
+    auto startPosition = 0;
+    auto inputString = input.str();
+    auto patternString = pattern_.str();
+    auto noTrailingWildcard = (patternString.back() != '%');
+
+    for (auto i = 0; i < reducedPatternLength_; i++) {
+      // Match the last fixed pattern with memcmp if there is no trailing '%'
+      // wildcard character.
+      if (noTrailingWildcard && i == reducedPatternLength_ - 1) {
+        auto fixedPatternLength = fixedPatterns_[i].size();
+        auto priorStartPosition = startPosition;
+        return (
+            input.size() - fixedPatternLength >= priorStartPosition &&
+            std::memcmp(
+                input.data() + input.size() - fixedPatternLength,
```

Comment:
input.data() + input.size() can be computed just once
pattern.data() + pattern.size() also can be computed just once.